app/controllers/nonsipp/memberdetails/RemoveMemberDetailsController.scala

```diff
@@ -72,11 +75,16 @@ class RemoveMemberDetailsController @Inject()(
           removeMemberDetails => {
             if (removeMemberDetails) {
               for {
-                updatedAnswers <- Future.fromTry(request.userAnswers.remove(MemberDetailsPage(srn, index)))
+                updatedAnswers <- softDeleteMember(srn, index).mapK[Future]
```

Comment:
I some cases, e.g, when the user start return in compile mode and still hasn't done the declaration, members should be hard deleted. Is there a logic to tell when to hard delete and when to soft delete?

Reply:
No logic like that, I can change the softDeletemember to just deleteMember and pass in a softDelete flag - is there a flag that is set when the the return is in compile or submitted mode?

Reply:
I think there is a need for that. Not sure about the exact conditions though. Maybe when the return main version is 001. There were some conversations about it, maybe Stuart can help.

Reply:
would you be ok if we open up a story for this and tackle it when I get back? should just be able to layer that functionality in quite nicely

Reply:
Yes, I'll create a story